### PR TITLE
New network architecture

### PR DESCRIPTION
**Pull request by alacuku**
# Description

This PR modifies the `liqo-gateway` component to use the new custom network namespace. Here we use all work done in the following PRs: #687, #686, #679, #673, #664,  #655, and #635 .

Fixes #(issue)

# How Has This Been Tested?

Existing end to end tests validate the new architecture.

### CODE REVIEW

File: cmd/liqonet/main.go [hunk 1]
```diff
@@ -59,7 +67,7 @@ func main() {
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
 	flag.StringVar(&runAs, "run-as", "tunnel-operator",
-		"The accepted values are: liqo-gateway, liqo-route, tunnelEndpointCreator-operator. The default value is \"tunnel-operator\"")
+		"The accepted values are: liqo-gateway, liqo-route, tunnelEndpointCreator-operator. The default value is \"liqo-gateway\"")
```
Comment (palexster) [inline cmd/liqonet/main.go:70]: I think this is an unwanted change.

File: cmd/liqonet/main.go [hunk 2]
```diff
@@ -72,64 +80,93 @@ func main() {
...
+			os.Exit(1)
+		}
+		vrm, err := liqorouting.NewVxlanRoutingManager(liqoconst.RoutingTableID, podIP.String(), liqoconst.OverlayNetPrefix, vxlanDevice)
+		if err != nil {
+			klog.Errorf("an error occurred while creating the vxlan routing manager: %v", err)
+			os.Exit(1)
+		}
+		er := mgr.GetEventRecorderFor(liqoconst.LiqoRouteOperatorName + "." + podIP.String())
+		rc := route_operator.NewRouteController(podIP.String(), vxlanDevice, vrm, er, mgr.GetClient())
+		if err != nil {
```
Comment (palexster) [inline cmd/liqonet/main.go:113]: which err?

File: cmd/liqonet/main.go [hunk 3]
```diff
@@ -72,64 +80,93 @@ func main() {
...
+			klog.Errorf("an error occurred while creating vxlan device : %v", err)
+			os.Exit(1)
+		}
+		vrm, err := liqorouting.NewVxlanRoutingManager(liqoconst.RoutingTableID, podIP.String(), liqoconst.OverlayNetPrefix, vxlanDevice)
+		if err != nil {
+			klog.Errorf("an error occurred while creating the vxlan routing manager: %v", err)
+			os.Exit(1)
+		}
+		er := mgr.GetEventRecorderFor(liqoconst.LiqoRouteOperatorName + "." + podIP.String())
+		rc := route_operator.NewRouteController(podIP.String(), vxlanDevice, vrm, er, mgr.GetClient())
```
Comment (aleoli) [inline cmd/liqonet/main.go:112]: can we use longer variable names to improve code readability?

**Review by aleoli — APPROVED** [review]

**Review by palexster — APPROVED** [review]

File: cmd/liqonet/main.go (hunk 1 above)
Comment (alacuku) [reply cmd/liqonet/main.go:70]: That was intentional, but I forgot to change the default value of `run-as` from `tunnel-operator` to `liqo-gateway`. Nice catch!

File: cmd/liqonet/main.go (hunk 2 above)
Comment (alacuku) [reply cmd/liqonet/main.go:113]: No error to be checked, you're right.

**Pull request merged**